Mergin/version_viewer_dialog.py

```diff
@@ -0,0 +1,639 @@
+
+        self.zoom_selected_action = QAction(
+            QgsApplication.getThemeIcon("/mActionZoomToSelected.svg"), "Zoom To Selection", self
+        )
+        self.zoom_selected_action.triggered.connect(self.zoom_selected)
+
+        self.toolbar.addAction(self.zoom_selected_action)
+
+        btn_add_changes = QPushButton("Add to project")
+        btn_add_changes.setIcon(QgsApplication.getThemeIcon("/mActionAdd.svg"))
+        menu = QMenu()
+        add_current_action = menu.addAction(QIcon(icon_path("file-plus.svg")), "Add current changes layer to project")
+        add_current_action.triggered.connect(self.add_current_to_project)
+        add_all_action = menu.addAction(QIcon(icon_path("folder-plus.svg")), "Add all changes layers to project")
+        add_all_action.triggered.connect(self.add_all_to_project)
+        btn_add_changes.setMenu(menu)
+
+        self.toolbar.addWidget(btn_add_changes)
+        self.toolbar.setIconSize(iface.iconSize())
+
+        self.map_canvas.enableAntiAliasing(True)
+        self.map_canvas.setSelectionColor(QColor(Qt.cyan))
+        self.pan_tool = QgsMapToolPan(self.map_canvas)
+        self.map_canvas.setMapTool(self.pan_tool)
+
+        self.current_diff = None
+        self.diff_layers = []
+        self.filter_model = None
+        self.layer_list.currentRowChanged.connect(self.diff_layer_changed)
+
+        self.icons = {
+            "added": "plus.svg",
+            "removed": "trash.svg",
+            "updated": "pencil.svg",
+            "renamed": "pencil.svg",
+            "table": "table.svg",
+        }
+        self.model_detail = QStandardItemModel()
+        self.model_detail.setHorizontalHeaderLabels(["Details"])
+
+        self.details_treeview.setEditTriggers(QAbstractItemView.NoEditTriggers)
+        self.details_treeview.setModel(self.model_detail)
+
+        self.versionModel.current_version = self.mp.version()
+
+    def exec(self):
+
+        try:
+            ws_id = self.mp.workspace_id()
+        except ClientError as e:
+            QMessageBox.warning(None, "Client Error", str(e))
+            return
+
+        # check if user has permissions
+        try:
+            usage = self.mc.workspace_usage(ws_id)
+            if not usage["view_history"]["allowed"]:
+                QMessageBox.warning(
+                    None, "Upgrade required", "To view the project history, please upgrade your subscription plan."
+                )
+                return
+        except ClientError:
+            # Some versions e.g CE, EE edition doesn't have
+            pass
+        super().exec()
+
+    def closeEvent(self, event):
+        self.save_splitters_state()
+        QDialog.closeEvent(self, event)
+
+    def save_splitters_state(self):
+        settings = QSettings()
+        settings.setValue("Mergin/VersionViewerSplitterSize", self.splitter_map_table.saveState())
+        settings.setValue("Mergin/VersionViewerSplitterVericalSize", self.splitter_vertical.saveState())
```

Comment:
```suggestion
        settings.setValue("Mergin/versionViewerSplitterSize", self.splitter_map_table.saveState())
        settings.setValue("Mergin/versionViewerSplitterVericalSize", self.splitter_vertical.saveState())
```
Settings entries are lowerCamelCase